parquet/src/main/java/org/apache/iceberg/parquet/ParquetVariantReaders.java

```diff
@@ -95,6 +96,19 @@ public static VariantValueReader objects(
         fieldReaders);
   }
 
+  public static VariantValueReader array(
+      int valueDefinitionLevel,
+      ParquetValueReader<?> valueReader,
+      int typedDefinitionLevel,
+      int repeatedDefinitionLevel,
+      int repeatedRepetitionLevel,
+      ParquetValueReader<?> elementReader) {
+    VariantValueReader typedReader =
+        new ListReader(
```

Comment:
The array reader looks good. I don't understand why this factory method is creating two readers though. I think that this should create the array reader and the caller should combine it with the other by calling `shredded`. That way, there are fewer values passed in here and it is more straightforward. We want to minimize passing DL, RL, or readers through the wrong args.